pootle/apps/pootle_app/management/commands/clear_stats.py

```diff
@@ -12,11 +12,45 @@
 # This must be run before importing Django.
 os.environ['DJANGO_SETTINGS_MODULE'] = 'pootle.settings'
 
-from pootle_app.management.commands import PootleCommand
+from optparse import make_option
 
+from django.core.management.base import NoArgsCommand
 
-class Command(PootleCommand):
-    help = "Flush stats cache."
+from django_redis import get_redis_connection
+from redis import WatchError
 
-    def handle_all_stores(self, translation_project, **options):
-        translation_project.clear_all_cache()
+from pootle.core.mixins.treeitem import POOTLE_DIRTY_TREEITEMS
+from pootle.core.models import Revision
+
+
+class Command(NoArgsCommand):
+    help = """Flush stats cache and dirty tree item counter and
+    rq data (optional).
+    """
+    shared_option_list = (
+        make_option(
+            "--clear-rq",
+            action="store_true",
+            default=False,
+            help=(u"Clear all RQ data"),
+        ),
+    )
+    option_list = NoArgsCommand.option_list + shared_option_list
+
+    def handle_noargs(self, **options):
+        # Delete all stats cache data
+        r_con = get_redis_connection('stats')
+        r_con.flushdb()
+
+        # Clear all dirty tree items
+        r_con = get_redis_connection('redis')
+        r_con.delete(POOTLE_DIRTY_TREEITEMS)
+
+        clear_rq = options.pop("clear_rq", False)
+        if not clear_rq:
+            return
+
+        # Clear all rq data and restore Pootle revision value
+        r_con.flushdb()
+        from pootle_store.models import Unit
```

Comment:
This is a stupid nit, but when I import something not at the top of the module I usually put it before the code block where it is being used and add a blank line afterwards to separate imports from code, so:

```python
        # Clear all rq data and restore Pootle revision value
        from pootle_store.models import Unit

        r_con.flushdb()
        Revision.set(Unit.max_revision())
```